service/src/main/kotlin/no/nav/su/se/bakover/service/statistikk/RevurderingStatistikkMapper.kt

```diff
@@ -54,6 +56,47 @@ internal class RevurderingStatistikkMapper(private val clock: Clock) {
         }
     }
 
+    internal object ResultatOgBegrunnelseMapper {
+        private const val innvilget = "Innvilget"
+        private const val opphørt = "Opphørt"
+        private const val ingenEndring = "Uendret"
+        private const val ingenEndringBegrunnelse = "Mindre enn 10% endring i inntekt"
+
+        private val stans = "Stans"
+        private val gjenopptak = "Gjenopptak"
+
+        internal data class ResultatOgBegrunnelse(
+            val resultat: String,
+            val begrunnelse: String?,
+        )
+
+        internal fun map(revurdering: Revurdering): ResultatOgBegrunnelse = when (revurdering) {
+            is IverksattRevurdering.Innvilget -> ResultatOgBegrunnelse(innvilget, null)
+            is IverksattRevurdering.Opphørt -> ResultatOgBegrunnelse(opphørt, listUtOpphørsgrunner(revurdering.utledOpphørsgrunner()))
+            is IverksattRevurdering.IngenEndring -> ResultatOgBegrunnelse(ingenEndring, ingenEndringBegrunnelse)
+            else -> throw ManglendeStatistikkMappingException(this, revurdering::class.java)
+        }
+
+        fun map(stansAvYtelse: StansAvYtelseRevurdering.IverksattStansAvYtelse) = ResultatOgBegrunnelse(stans, stansAvYtelse.revurderingsårsak.årsak.hentGyldigStansBegrunnelse())
+        fun map(gjenopptakAvYtelse: GjenopptaYtelseRevurdering.IverksattGjenopptakAvYtelse) = ResultatOgBegrunnelse(gjenopptak, gjenopptakAvYtelse.revurderingsårsak.årsak.hentGyldigGjenopptakBegrunnelse())
+
+        private fun listUtOpphørsgrunner(opphørsgrunner: List<Opphørsgrunn>): String = opphørsgrunner.joinToString(",")
+        private fun Revurderingsårsak.Årsak.hentGyldigStansBegrunnelse() = when (this) {
+            Revurderingsårsak.Årsak.MANGLENDE_KONTROLLERKLÆRING -> "Manglende kontrollerklæring"
+            else -> throw RuntimeException("$this er ikke en gyldig årsak for stans av ytelse")
```

Comment:
Vi kaller vel ikke statistikk-steget transaksjonelt? Så hvis vi først har kommet hit har vel originalkallet i sin helhet blitt utført? Burde vi ha flyttet statistikk ut i sin egen jobb, slik at det ikke kan påvirke the important stuff?

Reply:
Jeg tror jeg nevnte det i en tidligere pull-request også, men hvorfor kan `StansAvYtelseRevurdering.IverksattStansAvYtelse` og  `GjenopptaYtelseRevurdering.IverksattGjenopptakAvYtelse` inneholde revurderingsårsaker som _ikke_ kan oppstå?

Reply:
Det føles som `Revurderingsårsak.Årsak` er en potetklasse?

Reply:
Kommer i neste PR. 👍